tests/processing/strategies/test_streaming.py

```diff
@@ -505,3 +565,58 @@ def test_parallel_transform_step_terminate_workers() -> None:
         starting_processes,
     ), assert_changes(lambda: int(next_step.terminate.call_count), 0, 1):
         transform_step.terminate()
+
+
+def test_parallel_transform_step_bad_messages() -> None:
```

Comment:
I couldn't figure out why, but moving this test on top of `test_parallel_transform_step` breaks that test locally (it raises a `ChildProcessTerminated` exception). I didn't test on CI but maybe someone has an idea as to why that could be?

I'm guessing there's some cleanup I'm not doing that affects the next test but I'm not sure what it could be.

Reply:
This is concerning. Where is the `ChildProcessTerminated` get raised?
It may well be that, in the case where the parallel transform step raises an exception, something wrong happens in when sending the exception back from the subprocess to the main process.
We need to ensure that a subprocess does not terminate when raising the exception if you need to reuse it.

Reply:
The issue was I was calling `.join()` before `.close()`. Inverting the calls to those methods fixed the issue, no exception occurs regardless of where I position the test.